_posts/2019-08-06-serverless-static-sites.md

```diff
@@ -0,0 +1,232 @@
+
+Create a repository on GitHub(or your self hosted GitLab) called `example-site`. Add this repository to your GitHub app or add the `openfaas-cloud` tag if you are using GitLab.  
+Since we don't do recursive clones in OpenFaaS Cloud you will need to clone your site's theme instead of adding it as a submodule. Once you've done that commit and push your changes.  
+After a while, if you head over to your personal dashboard(find it at `system.<your domain>/dashboard/<username>`) you will see your new function deployed.  
+Edit the `FunctionIngress` so that it points to our brand new function `kubectl edit -n openfaas functioningresses.openfaas.com example-site`:
+```yaml
+apiVersion: openfaas.com/v1alpha2
+kind: FunctionIngress
+metadata:
+  name: example-site-tls
+  namespace: openfaas
+spec:
+  domain: "example.matiaspan.dev"
+  function: "<username>-example-site"
+  ingressType: "nginx"
+  tls:
+    enabled: true
+    issuerRef:
+      name: "letsencrypt-prod"
+      kind: "Issuer"
+```
+* You need to update the `function` to match with the new one. OpenFaaS Cloud deploys function using the following convention: `<username>-<function name>`. If you are unsure about the name you can find it with `faas-cli list`.
+
+We won't be using our previous function anymore so remove it:
+```sh
+faas-cli remove example-site
+```
+
+Try creating a new blog post with: `hugo new posts/my-first-post.md`. Remember to run that command inside the folder of your hugo site, not the root folder of the project.  
+Commit and push your changes again, after OpenFaaS Cloud does its thing you will be able to see your new changes deployed. Awesome!
+
+## CI/CD with GitLab
+If you can't or don't want to install OpenFaaS Cloud but you still want to have a [Netlify](https://netlify.com)-like experience then you can host your repository on GitLab and use their free CI/CD offerings. Save the following to a YAML file in the root folder of the repo `.gitlab-ci.yml`:
+```yaml
+image: docker:stable
+
+stages:
+  - build
+
+services:
+   - docker:dind
+
+before_script:
+  - apk add --no-cache git
+  - if [ -f "./faas-cli" ] ; then cp ./faas-cli /usr/local/bin/faas-cli || 0 ; fi
+  - if [ ! -f "/usr/local/bin/faas-cli" ] ; then apk add --no-cache curl git && curl -sSL cli.openfaas.com | sh && chmod +x /usr/local/bin/faas-cli && /usr/local/bin/faas-cli template pull && cp /usr/local/bin/faas-cli ./faas-cli ; fi
+
+build:
+  stage: build
+  script:
+    - git submodule init && git submodule update
+    # Build Docker image
+    - /usr/local/bin/faas-cli template pull https://github.com/matipan/openfaas-hugo-template
+    - /usr/local/bin/faas-cli build --tag=sha --parallel=2 -f example-site.yml
```

Comment:
`--parallel=2` - why?

Reply:
Copied over from a deployment where I have two functions and wanted to bulid them in parallel. I'll remove it since in this case is not necessary